src/analysis/sensitivity.jl

```diff
@@ -196,6 +201,80 @@ function pawn(
     return pawn(rs.inputs, y; S=S)
 end
 
+"""
+    convergence(X::DataFrame, y::NamedDimsArray, target_factors::Vector{Symbol}; n_steps::Int64=10)
+
+    Calculates the PAWN sensitivity index for an increasing number of scenarios where the maximum
+        is the total number of scenarios in scens. Number of scenario subsets determined by N_steps.
+
+# Arguments
+- `X` : Model inputs
+- `y` : Model outputs
+- `target_factors` : Names of target factors represented by columns in `X`
+- `n_steps` : Number of steps to cut the total number of scenarios into.
+
+# Returns
+NamedDimsArray, of min, mean, median, max, std, and cv summary statistics for an increasing
+number of scenarios.
+"""
+function convergence(
+    X::DataFrame,
+    y::NamedDimsArray,
+    target_factors::Vector{Symbol};
+    n_steps::Int64=10,
+)
+    N = length(y.scenarios)
+    step_size = floor(Int64, N / n_steps)
+    N_it = collect(step_size:step_size:N)
+
+    pawn_store = NamedDimsArray(
+        Array{Float64}(zeros(length(target_factors), 8, length(N_it)));
+        factors=target_factors,
+        Si=[:min, :lb, :mean, :median, :ub, :max, :std, :cv],
+        n_scenarios=N_it,
+    )
+    scens_idx = randperm(N)
+
+    for nn in N_it
+        pawn_store(; n_scenarios=nn) .= col_normalize(
+            pawn(X[scens_idx[1:nn], :], Array(y[scens_idx[1:nn]])),
```

Comment:
Could the method be passed in as an argument so we can use any sensitivity analysis method in the form of `f(X, y) -> Si`?

Reply:
I can do this, I'll just have to add lower and upper quantiles as outputs for the other methods

Reply:
Hi @ConnectedSystems , is the plan to allow only other "single output" metrics like pawn, or also methods like outcome_map and rsa which have different values for different ranges? If we want the latter I'll have to generalise the storage and some other things so let me know.

Reply:
No just methods that return a scalar (or set of scalars), thank you